plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Repository.java

```diff
@@ -61,6 +65,49 @@
 
     public static final String TYPE = "s3";
 
+    // prefix for aws client settings
+    private static final String PREFIX = "aws.config.";
```

Comment:
So if I understand what you are proposing here we would just support repository settings per repository (like `bucket`) but no more with global settings (like `repositories.s3.bucket`). Is that what you meant?

I looked at S3 repository specific settings we have today and I think it makes sense to have that in a single place, per repository.

If so, it would mean that we don't use anymore any `repositories.s3.*` settings. In that case, I'd suggest using `repositories.s3.client` to define the client settings for the `repository-s3` plugin and move all our classes under `org.elasticsearch.repositories.s3` package name. 

Client would go in `org.elasticsearch.repositories.s3.client` package.

WDYT?